src/EditorFeatures/Core/GoToDefinition/GoToDefinitionHelpers.cs

```diff
@@ -2,64 +2,46 @@
-        }
-
-        public static async Task<INavigableLocation?> GetDefinitionLocationAsync(
-            ISymbol symbol,
-            Solution solution,
-            IThreadingContext threadingContext,
-            IStreamingFindUsagesPresenter streamingPresenter,
-            CancellationToken cancellationToken,
-            bool thirdPartyNavigationAllowed = true)
-        {
-            var title = string.Format(EditorFeaturesResources._0_declarations,
-                FindUsagesHelpers.GetDisplayName(symbol));
-
-            var definitions = await GoToDefinitionFeatureHelpers.GetDefinitionsAsync(
-                symbol, solution, thirdPartyNavigationAllowed, cancellationToken).ConfigureAwait(false);
+        var location = await GetDefinitionLocationAsync(
+            symbol, solution, threadingContext, streamingPresenter, cancellationToken, thirdPartyNavigationAllowed).ConfigureAwait(false);
+        return await location.TryNavigateToAsync(
+            threadingContext, new NavigationOptions(PreferProvisionalTab: true, ActivateTab: true), cancellationToken).ConfigureAwait(false);
+    }
 
-            return await streamingPresenter.GetStreamingLocationAsync(
-                threadingContext, solution.Workspace, title, definitions, cancellationToken).ConfigureAwait(false);
-        }
+    public static async Task<INavigableLocation?> GetDefinitionLocationAsync(
+        ISymbol symbol,
+        Solution solution,
+        IThreadingContext threadingContext,
+        IStreamingFindUsagesPresenter streamingPresenter,
+        CancellationToken cancellationToken,
+        bool thirdPartyNavigationAllowed = true)
+    {
+        var title = string.Format(EditorFeaturesResources._0_declarations,
+            FindUsagesHelpers.GetDisplayName(symbol));
 
-        public static async Task<IEnumerable<INavigableItem>?> GetDefinitionsAsync(Document document, int position, CancellationToken cancellationToken)
-        {
-            // Try IFindDefinitionService first. Until partners implement this, it could fail to find a service, so fall back if it's null.
-            var findDefinitionService = document.GetLanguageService<IFindDefinitionService>();
-            if (findDefinitionService != null)
-            {
-                return await findDefinitionService.FindDefinitionsAsync(document, position, cancellationToken).ConfigureAwait(false);
-            }
+        var definitions = await GoToDefinitionFeatureHelpers.GetDefinitionsAsync(
+            symbol, solution, thirdPartyNavigationAllowed, cancellationToken).ConfigureAwait(false);
 
-            // Removal of this codepath is tracked by https://github.com/dotnet/roslyn/issues/50391. Once it is removed, this GetDefinitions method should
-            // be inlined into call sites.
-            var goToDefinitionsService = document.GetRequiredLanguageService<IGoToDefinitionService>();
```

Comment:
this was required, which didn't work if hte language didn't export either of these.